sktime/performance_metrics/forecasting/_functions.py

```diff
@@ -135,3 +139,109 @@ def mape_loss(y_test, y_pred):
+    forecaster : sktime.forecaster
+        Any forecaster
+    cv : sktime.SlidingWindowSplitter or sktime.ExpandingWindowSplitter
+        Splitter of how to split the data into test data and train data
+    y : pd.Series
+        Target time series to which to fit the forecaster.
+    X : pd.DataFrame, optional (default=None)
+        Exogenous variables (ignored)
+    strategy : str, optional
+        Must be "refit" or "update", by default "refit". The strategy defines
+        whether forecaster is only fitted on the first train window data and
+        then updated or always refitted.
+    scoring : function, optional
+        A score function that takes y_pred and y_test as arguments,
+        by default smape_loss
+    return_data : bool, optional
+        Returns three additional columns in the DataFrame, by default False.
+        The cells of the columns contain each a pd.Series for y_train,
+        y_pred, y_test.
+
+    Returns
+    -------
+    pd.DataFrame
+        DataFrame that contains several columns with information regarding each
+        refit/update and prediction of the forecaster.
+
+    Examples
+    --------
+    >>> from sktime.datasets import load_airline
+    >>> from sktime.performance_metrics.forecasting import evaluate
+    >>> from sktime.forecasting.model_selection import ExpandingWindowSplitter
+    >>> from sktime.forecasting.naive import NaiveForecaster
+    >>> y = load_airline()
+    >>> forecaster = NaiveForecaster(strategy="drift", sp=12)
+    >>> cv = ExpandingWindowSplitter(
+        initial_window=24,
+        step_length=12,
+        fh=[1,2,3,4,5,6,7,8,9,10,11,12]
+        )
+    >>> evaluate(forecaster=forecaster, y=y, cv=cv)
+    """
+    assert cv.initial_window is not None, "cv must have an initial_window"
+    assert isinstance(evaluate, types.FunctionType), "scoring must be a function"
+    assert strategy in ["refit", "update"], "strategy must be either fit or update"
+
+    n_splits = cv.get_n_splits(y)
+    results = pd.DataFrame()
+
+    for i, (train, test) in enumerate(tqdm(cv.split(y), total=n_splits)):
+        # workaroud to avoid training on smaller windows
+        if len(train) >= cv.initial_window:
+            # create train/test data
+            y_train = y.iloc[train]
+            y_test = y.iloc[test]
+            if X:
+                X_train = X.iloc[train]
+                X_test = X.iloc[test]
+            else:
+                X_train = None
+                X_test = None
+
+            # fit/update
+            start_fit = time.time()
+            if strategy == "refit" or i == 0:
+                forecaster.fit(y=y_train, X=X_train)
+            elif strategy == "update" and i != 0:
+                forecaster.update(y=y_train, X=X_train)
+            fit_time = time.time() - start_fit
+
+            # predict
+            start_pred = time.time()
+            y_pred = forecaster.predict(
+                fh=[x + 1 for x in range(len(y_test))], X=X_test
```

Comment:
hm, should this not be `fh = ForecastingHorizon(y_test.index, is_relative=False)` instead?
To avoid issues with non-integer indexed data.